src/components/Snippet/FileManager.vue

```diff
@@ -0,0 +1,215 @@
+<script>
```

Comment:
> The access control will be using this as well

What do you mean by that?  Nothing within the library should be importing FileManager.

Can move this into components if you want, I just though it would be good to separate so other components don't try to import from snippet/.

Reply:
I mean, the FileManager is not a one-off component, more feature will be added to it. Such as the Access Control in the future PR, so it is a core component that deserves to live in the components folder?

Reply:
Sure, so FileManager may import other things, but other things will never import it.  It's a one way relationship, which is why I separated it, just like `views/` and `components/` are separated in regular projects.

It's easy to remember not to import FileManager when it's just one thing, but if there were 6, it's nice to have the directory separation as a red flag during code review.

Reply:
Oh, I see what you mean. You meant that anything under component is `reusable` component, this snippet is on the contrary, final. 
Make sense, let's keep it in the snippet.